src/NServiceBus.Core/IBusExtensions.cs

```diff
@@ -94,7 +94,7 @@ public static void Subscribe<T>(this IBus bus)
         {
             Guard.AgainstNull("bus", bus);
 
-            bus.Subscribe(typeof(T), new SubscribeOptions());
+            bus.SubscribeAsync(typeof(T), new SubscribeOptions());
```

Comment:
We need to fix this. The synchronous version should be obsoleted and only the async version should be exposed